base/logging.go

```diff
@@ -236,6 +237,19 @@ func SetLogLevel(level int) {
 	logLevel = level
 }
 
+func SetRedaction(redactionLevel RedactionLevel) {
+	Logf("Log redaction level: %s", redactionLevel)
+	switch redactionLevel {
+	case RedactFull:
+		RedactUserData = true
+	case RedactPartial:
+		RedactUserData = true
+	case RedactNone:
+	default:
+		RedactUserData = false
+	}
+}
+
```

Comment:
Add a unit test that loads from config and validates RedactUserData ends up set properly.